manifests/charts/gateway/templates/deployment.yaml

```diff
@@ -68,7 +68,7 @@ spec:
             allowPrivilegeEscalation: false
             privileged: false
             readOnlyRootFilesystem: true
-            {{- if not (eq .Values.platform "openshift") }}
+            {{- if not (eq .Values.global.platform "openshift") }}
```

Comment:
The `gateway` chart does not use global at all, much like ztunnel does not. If we want to do this, we should do https://github.com/istio/istio/pull/51749/.

The background here...
* `global` is for composing helm charts with a parent chart and subchart
* Istio used to just have an 'istio' chart, then decomposed them. For backwards compat (this was a mistake) we kept the `global.` and `pilot.` prefixes everywhere. 
* Helm, in what I consider a design flaw, makes individual charts change to support composing them (which is a layering violation). #51749 essentially fixes this by flattening globals so they can transparently be used.

Reply:
Ok, I see it. If I do the change to instead of using `global.platorm` to use `platform` it will work also and we will avoid this thing with Helm, so the values can be directly used without the need to set `global.platform`

Reply:
It's not a layering violation - Helm charts do not layer, they compose - and all composition involves (some amount of) global data.

We may wish Helm layered instead of composed, but it does not - we can fight about it or accept it.

Globals are a documented and supported part of Helm and there's nothing inherently wrong with using them.

`platform` _is already_ a global in current code. This PR is not making it a global, it is exposing the fact that it is and has been for some time.

If we want to de-globalize `platform` I am not against that, but I would point out it is used across charts and does not vary, which makes it a pretty reasonable candidate for a global. Since it already is a global, I don't object to it staying that way.

Reply:
It is already `global.platform` everywhere else but here and `ztunnel` AFAICT - the API ls _already_ broken.

We can unbreak it by making it `platform` everywhere, or unbreak it by making it `global.platform` everywhere - I I'm indifferent to which and don't see an important distinction.

So my vote is the simplest and least invasive form of unbreaking - if that's `use global.platform here` - then that's fine.

If it's `use platform everywhere else` - then that's fine too. I'll leave that to the PR author.

Reply:
Its not broken at all, because Istio is shipping a bunch of independent charts. We could have called it `platform` in one, `notPlatform`, `icantbelieveitsnotplatform`, `global.platform`and that would be "fine" (sans bad naming). There is no expectation that a value in one chart has the same meaning or same configuration.

Reply:
> There is no expectation that a value in one chart has the same meaning or same configuration.

Strong disagree here. It's absurd to ask our users to remember that the "right" value for chart A is `foo.bar.x.y` and the "right" value for chart B is `baz.boo.q.c` when there is no actual technical or practical reason why this variation should exist, and they must install both charts to get a functional system.

It's just bad UX, and it makes our charts much messier and harder to maintain - case in point this, where we have functionally redundant values used in 3-way conditionals, for no particularly defensible reason.

Reply:
tl;dr - I'm not comfortable leaving it in the current broken status quo or involving @fjglira in yet another spat over Istio's unique helm opinions, so I'm bowing out of this review.

Reply:
Let me clarify: the charts, as currently designed and implemented, does not make that assumption. Whether you expect it or think users expect it is up to you/the users :slightly_smiling_face: